flows/interfaces.go

```diff
@@ -163,12 +163,17 @@ type Context interface {
 }
 
 type Event interface {
-	CreatedOn() *time.Time
+	CreatedOn() time.Time
 	SetCreatedOn(time.Time)
 
 	Step() StepUUID
 	SetStep(StepUUID)
 
+	FromCaller() bool
+	SetFromCaller(bool)
```

Comment:
Eric kind of has a use case for that actually in the broadcast batching.. being able to map which action created a `send_msg` event would actually be useful there I think.